charts/vector/templates/_helpers.tpl

```diff
@@ -241,9 +241,95 @@ Print the `url` option for the Vector command.
+
+To forward logs from Datadog Agents deployed with the {{ include "_fmt.yellow" "datadog" }} Helm chart,
+include the following in the {{ include "_fmt.yellow" "values.yaml" }} for your {{ include "_fmt.yellow" "datadog" }} chart:
+
+For Datadog Agents version {{ include "_fmt.yellow" "7.35" }}/{{ include "_fmt.yellow" "6.35" }} or greater:
+
+{{ include "_fmt.blue" "datadog:" }}
+  {{ include "_fmt.blue" "containerExclude:" }} "name:vector"
+  {{ include "_fmt.blue" "logs:" }}
+    {{ include "_fmt.blue" "enabled:" }} {{ include "_fmt.yellow" "true" }}
+    {{ include "_fmt.blue" "containerCollectAll:" }} {{ include "_fmt.yellow" "true" }}
+{{ include "_fmt.blue" "agents:" }}
+  {{ include "_fmt.blue" "useConfigMap:" }} {{ include "_fmt.yellow" "true" }}
+  {{ include "_fmt.blue" "customAgentConfig:" }}
+    {{ include "_fmt.blue" "kubelet_tls_verify:" }} {{ include "_fmt.yellow" "false" }}
+    {{ include "_fmt.blue" "vector:" }}
+      {{ include "_fmt.blue" "logs:" }}
+        {{ include "_fmt.blue" "enabled:" }} {{ include "_fmt.yellow" "true" }}
+        {{- if not .Values.haproxy.enabled }}
+        {{ include "_fmt.blue" "url:" }} "{{ $protocol }}://{{ include "vector.fullname" $ }}.{{ $.Release.Namespace }}:{{ $sourceDatadogAgentPort | default "8282" }}"
+        {{- else }}
+        {{ include "_fmt.blue" "url:" }} "{{ $protocol }}://{{ include "haproxy.fullname" $ }}.{{ $.Release.Namespace }}:{{ $sourceDatadogAgentPort | default "8282" }}"
+        {{- end }}
+      {{ include "_fmt.blue" "metrics:" }}
+        {{ include "_fmt.blue" "enabled:" }} {{ include "_fmt.yellow" "true" }}
+        {{- if not .Values.haproxy.enabled }}
+        {{ include "_fmt.blue" "url:" }} "{{ $protocol }}://{{ include "vector.fullname" $ }}.{{ $.Release.Namespace }}:{{ $sourceDatadogAgentPort | default "8282" }}"
+        {{- else }}
+        {{ include "_fmt.blue" "url:" }} "{{ $protocol }}://{{ include "haproxy.fullname" $ }}.{{ $.Release.Namespace }}:{{ $sourceDatadogAgentPort | default "8282" }}"
+        {{- end }}
+
+For Datadog Agents version {{ include "_fmt.yellow" "7.34" }}/{{ include "_fmt.yellow" "6.34" }} and older:
+
+{{ include "_fmt.blue" "datadog:" }}
+  {{ include "_fmt.blue" "containerExclude:" }} "name:vector"
+  {{ include "_fmt.blue" "logs:" }}
+    {{ include "_fmt.blue" "enabled:" }} {{ include "_fmt.yellow" "true" }}
+    {{ include "_fmt.blue" "containerCollectAll:" }} {{ include "_fmt.yellow" "true" }}
+{{ include "_fmt.blue" "agents:" }}
+  {{ include "_fmt.blue" "useConfigMap:" }} {{ include "_fmt.yellow" "true" }}
+  {{ include "_fmt.blue" "customAgentConfig:" }}
+    {{ include "_fmt.blue" "kubelet_tls_verify:" }} {{ include "_fmt.yellow" "false" }}
+    {{ include "_fmt.blue" "logs_config:" }}
+      {{- if not .Values.haproxy.enabled }}
+      {{ include "_fmt.blue" "logs_dd_url:" }} "{{ include "vector.fullname" $ }}.{{ $.Release.Namespace }}:{{ $sourceDatadogAgentPort | default "8282" }}"
```

Comment:
I don't believe so, all of these options should be at the same indention depth - _honestly_ we should be using `nindent` or `indent` to provide whitespace via the templating engine, rather than my including literal spaces at the beginning of these lines. However the templating engine was getting angry and I got tired of fighting with it, so I settled for this easier but more brittle/less idiomatic way.

If we do a second pass on this to clean up the templates and make things more re-usable functions we should be able to use the indention functions more easily.